src/Tools/GenerateDocumentationAndConfigFiles/Program.cs

```diff
@@ -532,7 +579,9 @@ static bool checkHelpLink(string helpLink)
             string? category,
             string? customTag,
             SortedList<string, DiagnosticDescriptor> sortedRulesById,
-            string analyzerPackageName)
+            string analyzerPackageName,
+            bool validateOnly,
+            ref bool validationResult)
```

Comment:
What about returning a boolean value instead of having this `ref` parameter?

Reply:
@Evangelink This will require checking this return value in multiple places and update the Main local variable based on the return result. I wanted to avoid adding this duplicate logic. Do you see other solution that avoids ref and also avoids duplicating this logic?